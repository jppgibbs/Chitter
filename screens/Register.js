import React, {Component} from 'react';
import {
  Text,
  TextInput,
  View,
  Alert,
  Button,
  FlatList,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';

class CreateAccount extends Component {
  // Construct variables with default empty values
  constructor(props) {
    super(props);
    this.state = {
      given_name: '',
      family_name: '',
      email: '',
      password: '',
      user_id: '',
      validation: '',
    };
  }

  createAccount() {
    let res1 = JSON.stringify({
      given_name: this.state.given_name,
      family_name: this.state.family_name,
      email: this.state.email,
      password: this.state.password,
    });

    console.log(res1);

    return fetch('http://10.0.2.2:3333/api/v0.0.5/user', {
      method: 'POST',
      body: res1,
      headers: {
        'Content-Type': 'application/json',
      },
    })
      .then(response => {
        if (response.status === 201) {
          Alert.alert('Account Created', 'Log in with your new account!');
          this.props.navigation.navigate('Account');
        } else {
          Alert.alert(
            'Creation Failed',
            'Account creation unsuccessful. Check your details.',
          );
        }
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    return (
      <View style={styles.viewStyle} accessible={true}>
        <Text style={styles.title} accessibilityRole="text">
          Create Account
        </Text>
        <Text style={styles.bodyText} accessibilityRole="text">
          First Name
        </Text>
        <TextInput
          style={styles.textEntry}
          onChangeText={text => this.setState({given_name: text})}
          value={this.state.given_name}
          placeholderTextColor="#918f8a"
          placeholder="John"
          accessibilityComponentType="none"
          accessibilityRole="none"
          accessibilityLabel="First Name"
          accessibilityHint="Enter your first name"
        />
        <Text style={styles.bodyText} accessibilityRole="text">
          Second Name
        </Text>
        <TextInput
          style={styles.textEntry}
          onChangeText={text => this.setState({family_name: text})}
          value={this.state.family_name}
          placeholderTextColor="#918f8a"
          placeholder="Smith"
          accessibilityComponentType="none"
          accessibilityRole="none"
          accessibilityLabel="Second Name"
          accessibilityHint="Enter your second name"
        />
        <Text style={styles.bodyText} accessibilityRole="text">
          Email
        </Text>
        <TextInput
          style={styles.textEntry}
          onChangeText={text => this.setState({email: text})}
          value={this.state.email}
          textContentType="emailAddress"
          placeholderTextColor="#918f8a"
          placeholder="example@example.com"
          accessibilityComponentType="none"
          accessibilityRole="none"
          accessibilityLabel="Enter email"
          accessibilityHint="Enter the email you wish to use for your account"
        />
        <Text style={styles.bodyText} accessibilityRole="text">
          Password
        </Text>
        <TextInput
          style={styles.textEntry}
          onChangeText={text => this.setState({password: text})}
          value={this.state.password}
          secureTextEntry
          placeholderTextColor="#918f8a"
          placeholder="Password"
          accessibilityComponentType="none"
          accessibilityRole="none"
          accessibilityLabel="Enter password"
          accessibilityHint="Enter the password you wish to use for your account"
        />
        <TouchableOpacity
          onPress={() => this.createAccount()}
          style={styles.button}>
          <Text style={styles.bodyText} accessibilityRole="text">
            Create Account
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  viewStyle: {
    justifyContent: 'center',
    flex: 1,
    backgroundColor: '#17202b',
    color: '#ffffff',
  },
  button: {
    alignItems: 'center',
    elevation: 2,
    padding: 10,
    marginTop: 5,
    marginBottom: 0,
    marginLeft: 15,
    marginRight: 15,
    borderColor: '#101010',
    borderWidth: 1,
    borderRadius: 2,
    backgroundColor: '#2296f3',
  },
  textEntry: {
    alignItems: 'center',
    padding: 5,
    color: '#ffffff',
    marginTop: 5,
    marginBottom: 0,
    borderColor: '#2296f3',
    borderRadius: 2,
    borderWidth: 1,
    backgroundColor: '#273341',
    elevation: 3,
    marginLeft: 15,
    marginRight: 15,
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#ffffff',
    marginLeft: 15,
    marginRight: 15,
    marginBottom: 5,
  },
  bodyText: {
    color: '#ffffff',
    marginLeft: 15,
    marginRight: 15,
  },
});

export default CreateAccount;
