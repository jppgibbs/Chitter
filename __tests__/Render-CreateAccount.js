import 'react-native';
import React from 'react';
import Register from '../screens/Register.js';
import renderer from 'react-test-renderer';

test('Does create account render?', () => {
  const snap = renderer.create(<Register />).toJSON();

  expect(snap).toMatchSnapshot();
});
