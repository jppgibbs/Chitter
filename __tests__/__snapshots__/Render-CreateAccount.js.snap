// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create render 1`] = `
<View
  accessible={true}
  style={
    Object {
      "backgroundColor": "#17202b",
      "color": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessibilityRole="text"
    accessible={true}
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Create Account
  </Text>
  <Text
    accessibilityRole="text"
    accessible={true}
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    First Name
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter your first name"
        accessibilityLabel="First Name"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="John"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Second Name
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Regular",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "400",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter your second name"
        accessibilityLabel="Second Name"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="  Smith"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Email
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Regular",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "400",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter the email you wish to use for your account"
        accessibilityLabel="Enter email"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="  example@example.com"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        textContentType="emailAddress"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Password
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter the password you wish to use for your account"
        accessibilityLabel="Enter password"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="  Password"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessibilityHint="Press this to create your account"
      accessibilityLabel="Create Account"
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Create Account
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Does create account render? 1`] = `
<View
  accessible={true}
  style={
    Object {
      "backgroundColor": "#17202b",
      "color": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessibilityRole="text"
    accessible={true}
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Create Account
  </Text>
  <Text
    accessibilityRole="text"
    accessible={true}
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    First Name
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter your first name"
        accessibilityLabel="First Name"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="John"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Second Name
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Regular",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "400",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter your second name"
        accessibilityLabel="Second Name"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="  Smith"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Email
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Regular",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "400",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter the email you wish to use for your account"
        accessibilityLabel="Enter email"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="  example@example.com"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        textContentType="emailAddress"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    Password
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        accessibilityComponentType="none"
        accessibilityHint="Enter the password you wish to use for your account"
        accessibilityLabel="Enter password"
        accessibilityRole="none"
        accessible={true}
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="  Password"
        placeholderTextColor="#918f8a"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessibilityHint="Press this to create your account"
      accessibilityLabel="Create Account"
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Create Account
        </Text>
      </View>
    </View>
  </View>
</View>
`;
